### DISCUSSION TITLE
Make use of `color` type

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2721] [Suggestions]
Throughout Wiremod and E2, `vector` is often used as a replacement for a `color`.

This works but leads to pollution of methods / functions in E2 for color-based things that take vectors, and performance hits since `Vector`s need to be converted to `Color` structs before being used by GLua.

[There's already an entry in `WireLib.DT` for `Color`, so why hasn't it been used?](https://github.com/wiremod/wire/blob/ccfd65ee6bf597e8d2349da78c2e13362284f6ad/lua/wire/server/wirelib.lua#L95)

So I propose adding a `color` type to E2, and potentially to use it in place of `vector` in certain wire components with backwards compatibility if possible.

**Comment by thegrb93** [comment]
Backwards compatibility with existing e2s will be the issue here.

**Reply by vurvdev** [reply]
I don't see how there'd be any backwards compatibility issues E2 wise since you can just make an overload that takes a `color` instead of a `vector` or `vector4` for certain functions, and deprecate the ones taking `vector`s.

Now wiremod yes, if we try and say replace the light color input to take a `color` instead of `vector`. Although I think a system to allow multiple types of input would be possible for backwards compat. Just kinda hard

**Comment by vurvdev** [comment]
I also forgot to mention that `vector4`'s sole purpose is practically to act as a color, which is really weird, so could deprecate it almost entirely.
